lua/dotfyle_metadata/keymaps.lua

```diff
@@ -1,32 +1,51 @@
 ---Get all user defined keymaps
 ---@return table
+
+local function check_map(keymap)
+  -- check if the mapping is one we want to include
+
+  if  keymap.mode ~= "n" and keymap.mode ~= "i"
+  and keymap.mode ~= "x" and keymap.mode ~= "v"
+  then
+    return false
+  end
+  if string.find(keymap.lhs, "<Plug>")
+  or string.find(keymap.rhs, "<Plug>")
+  or string.find(keymap.lhs, "Þ")
+    then
+      return false
+  end
+
+  return true
+end
+
 return function()
 	local keymaps = {}
 	local global_keymaps = vim.api.nvim_get_keymap("")
 
 	for _, map in pairs(global_keymaps) do
 		-- translate the strings into something usable by :map
-		map.lhs = vim.api.nvim_replace_termcodes(map.lhs, true, true, false)
-
-		if map.rhs == "" then
-			-- replace the empty string with empty modifier
-			map.rhs = "<Nop>"
-		end
+		map.lhs = vim.fn.maparg(map.lhs, map.mode, false, true).lhs
```

Comment:
From your [comment before](https://github.com/creativenull/dotfyle-metadata.nvim/pull/4#issuecomment-1652640647), I find this misleading since ` w` and `w` have different meaning. We want to be able to change this to a modifier (in this case `<Space>`)

I personally won't use `maparg` if its obscuring `lhs` as it was typed and just stick with `nvim_replace_termcodes`.

Reply:
I'm not sure what you mean here. I the example case what I typed was `<leader>w` but what we get is ` w`

Reply:
We do get ` w`, what we don't currently get is the translation of the space character in to `<space>`. but with maparg we will